bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpUtil.java

```diff
@@ -147,13 +137,52 @@ public static String executeUrl(String httpMethod, String url, Properties httpHe
      * @param proxyUser the username to authenticate with the proxy
      * @param proxyPassword the password to authenticate with the proxy
      * @param nonProxyHosts the hosts that won't be routed through the proxy
+     *
      * @return the response body or <code>NULL</code> when the request went wrong
      * @throws IOException when the request execution failed, timed out or it was interrupted
      */
     public static String executeUrl(String httpMethod, String url, Properties httpHeaders, InputStream content,
             String contentType, int timeout, String proxyHost, Integer proxyPort, String proxyUser,
             String proxyPassword, String nonProxyHosts) throws IOException {
 
+        ContentResponse response = executeUrlAndGetReponse(httpMethod, url, httpHeaders, content, contentType, timeout,
+                proxyHost, proxyPort, proxyUser, proxyPassword, nonProxyHosts);
+        String encoding = response.getEncoding() != null ? response.getEncoding().replaceAll("\"", "").trim() : "UTF-8";
+        String responseBody;
+        try {
+            responseBody = new String(response.getContent(), encoding);
+            if (!responseBody.isEmpty()) {
+                logger.trace(responseBody);
```

Comment:
Quality tools will complain about that statement because the string is not constant.